src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-FSharp/Program.fs

```diff
@@ -1,27 +1,52 @@
 namespace Company.WebApplication1
-
+#nowarn "20"
 open System
 open System.Collections.Generic
 open System.IO
 open System.Linq
 open System.Threading.Tasks
 open Microsoft.AspNetCore
+open Microsoft.AspNetCore.Builder
 open Microsoft.AspNetCore.Hosting
+#if (!NoHttps)
+open Microsoft.AspNetCore.HttpsPolicy;
+#endif
 open Microsoft.Extensions.Configuration
+open Microsoft.Extensions.DependencyInjection
 open Microsoft.Extensions.Hosting
 open Microsoft.Extensions.Logging
 
 module Program =
     let exitCode = 0
 
-    let CreateHostBuilder args =
-        Host.CreateDefaultBuilder(args)
-            .ConfigureWebHostDefaults(fun webBuilder ->
-                webBuilder.UseStartup<Startup>() |> ignore
-            )
-
     [<EntryPoint>]
     let main args =
-        CreateHostBuilder(args).Build().Run()
+        let builder = WebApplication.CreateBuilder(args)
+
+        builder.Services.AddControllersWithViews().AddRazorRuntimeCompilation()
+        builder.Services.AddRazorPages()
+
+        let app = builder.Build()
+
+        if not(builder.Environment.IsDevelopment()) then
+            app.UseExceptionHandler("/Home/Error")
+
+#if (!NoHttps)
+            app.UseHsts() |> ignore // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
+
+        app.UseHttpsRedirection()
+#endif
+
+        app.UseStaticFiles()
+        app.UseRouting()
+        app.UseAuthorization()
+
+        app.MapControllerRoute(
+                name = "default",
```

Comment:
Too much indent here, use 4-space (maybe try auto-formatting with https://fsprojects.github.io/fantomas-tools/#/fantomas/preview - paste it all in and see how it changes)